cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -226,73 +222,76 @@ void WholeStageResultIterator::collectMetrics() {
       }
       // Special handing for Filter over Project case. Filter metrics are
       // omitted.
-      numOfStats += 1;
+      statsNum += 1;
       continue;
     }
-    numOfStats += planStats.at(nodeId).operatorStats.size();
+    statsNum += planStats.at(nodeId).operatorStats.size();
   }
 
-  metrics_ = std::make_shared<Metrics>(numOfStats);
-  int metricsIdx = 0;
+  metrics_ = std::move(std::make_unique<Metrics>(statsNum));
```

Comment:
Could you explain the difference between `metrics_ = std::move(std::make_unique<Metrics>(statsNum));` and `metrics_ = std::make_unique<Metrics>(statsNum);`?

My IDE always complain about this usage.

Reply:
the `unique_ptr` does not support `unique_ptr& operator=(unique_ptr& rhs)` .
the compiler will report errors without `std::move`.